qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -367,9 +367,54 @@ def get_spinZ_matrix_elements(mol_name, hf_data, n_active_electrons=None, n_acti
     return spinz_matrix_elements
 
 
+def particle_number(n_orbitals, mapping="jordan_wigner"):
+    r"""Computes the particle number operator :math:`\hat{N}=\sum_\alpha \hat{n}_\alpha`
+    in the Pauli basis.
+
+    The particle number operator reads,
+
+    .. math::
+
+        \hat{N} = \sum_\alpha \hat{c}_\alpha^\dagger \hat{c}_\alpha.
+
+    The index :math:`\alpha` runs over the basis of single-particle orbitals
+    :math:`\vert \alpha \rangle`. The operators :math:`\hat{c}^\dagger` and :math:`\hat{c}` are
+    the particle creation and annihilation operators, respectively.
+
+    Args:
+        n_orbitals (int): Number of orbitals. If an active space is defined, 'n_orbitals'
+            is the number of active orbitals.
+        mapping (str): Specifies the transformation to map the fermionic operator to the
+            Pauli basis. Input values can be ``'jordan_wigner'`` or ``'bravyi_kitaev'``.
+
+    Returns:
+        pennylane.Hamiltonian: the fermionic-to-qubit transformed observable
+
+    **Example**
+
+    >>> n_orbitals = 2
+    >>> N = particle_number(n_orbitals, mapping="jordan_wigner")
+    >>> print(N)
+    (2.0) [I0]
+    + (-0.5) [Z0]
+    + (-0.5) [Z1]
+    + (-0.5) [Z2]
+    + (-0.5) [Z3]
+    """
+
+    n_spin_orbs = 2 * n_orbitals
+
+    table = np.ones((n_spin_orbs, 3))
+    for alpha in range(n_spin_orbs):
+        table[alpha, :2] = alpha
```

Comment:
@agran2018, completely up to you, but you could avoid the `for` loop by noting the structure of the final table. For example, if `n_orbitals = 3`, then,

```pycon
>>> print(table)
array([[0., 0., 1.],
       [1., 1., 1.],
       [2., 2., 1.],
       [3., 3., 1.],
       [4., 4., 1.],
       [5., 5., 1.]])
```

That is, the first two _columns_ will always be filled with `np.arange(n_spin_orbs)`, and the remaining column will always be 1.

So we can encode this structure _directly_, which will be much more efficient:

```pycon
>>> r = np.arange(n_spin_orbs)
>>> table = np.vstack([r, r, np.ones([n_spin_orbs])]).T
print(table)
array([[0., 0., 1.],
       [1., 1., 1.],
       [2., 2., 1.],
       [3., 3., 1.],
       [4., 4., 1.],
       [5., 5., 1.]])
```

Reply:
Thank you @josh146!. I feel embarrassed any time I have a loop and I suspect it can be vectorized using Numpy. But definitely I have to learn to use this nice tools!.

Reply:
This is done.